hnn_core/tests/test_gui.py

```diff
@@ -422,6 +423,119 @@ def test_gui_adaptive_spectrogram():
     plt.close('all')
 
 
+def test_gui_layer2_dipole():
+    """Test the visualization of layer 2 dipole in the HNNGUI."""
+    gui = HNNGUI()
+    gui.compose()
+    gui.params['N_pyr_x'] = 3
+    gui.params['N_pyr_y'] = 3
+
+    gui.run_button.click()
+    figid = 1
+    figname = f'Figure {figid}'
+    axname = 'ax1'
+    gui._simulate_viz_action("edit_figure", figname, axname, 'default',
+                             'layer2 dipole', {}, 'clear')
+    gui._simulate_viz_action("edit_figure", figname, axname, 'default',
+                             'layer2 dipole', {}, 'plot')
+    # Check if data is plotted on the axes
+    assert len(gui.viz_manager.figs[figid].axes) == 2
+    # Check default figs have data on their axis
+    assert gui.viz_manager.figs[figid].axes[0].has_data()
```

Comment:
Looking good. Minor, but since we are only making changes to ax1. I think we can drop the check on axis[0] for these tests.